content/docs/handling-events.md

```diff
@@ -152,3 +152,21 @@ Inside a loop it is common to want to pass an extra parameter to an event handle
 The above two lines are equivalent, and use [arrow functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) and [`Function.prototype.bind`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Function/bind) respectively.
 
 In both cases, the `e` argument representing the React event will be passed as a second argument after the ID. With an arrow function, we have to pass it explicitly, but with `bind` any further arguments are automatically forwarded.
+
+### Event handlers with Currying
+
+We can also accomplish event handling by using function currying with arrow functions.  To do this we will need to slightly update our event handler. When the handler is invoked by the render function, the outer function is called which passes our arguments in.  The handler also returns an inner function.  The inner function is what is called when the given event is executed.  The inner function has access to the arguments in the outer function.
+
+```js
+class LoggingButton extends React.Component {
+  handleClick = (i) => (e) => {
+    console.log(`button ${i} was clicked`);
+  }
+
+  render() {
+    return (
+      <button onClick={this.handleClick(i)}Click me</button>
```

Comment:
Where does `i` come from? Please show a complete example (e.g. a list of buttons) to avoid confusion.

Also, this example is missing the closing ">" between `onClick` and "Click me"